floor/writeread_test.go

```diff
@@ -0,0 +1,228 @@
+package floor
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+	"time"
+
+	goparquet "github.com/fraugster/parquet-go"
+	"github.com/fraugster/parquet-go/parquet"
+	"github.com/fraugster/parquet-go/parquetschema"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func writeReadOne(t *testing.T, o interface{}, schema string) interface{} {
+	schemaDef, err := parquetschema.ParseSchemaDefinition(schema)
+	require.NoError(t, err)
+
+	var buf bytes.Buffer
+	w := NewWriter(goparquet.NewFileWriter(&buf,
+		goparquet.WithCompressionCodec(parquet.CompressionCodec_SNAPPY),
+		goparquet.WithSchemaDefinition(schemaDef),
+	))
+
+	err = w.Write(o)
+	require.NoError(t, err)
+
+	err = w.Close()
+	require.NoError(t, err)
+
+	fr, err := goparquet.NewFileReader(bytes.NewReader(buf.Bytes()))
+	require.NoError(t, err)
+
+	pr := NewReader(fr)
+
+	pr.Next()
+
+	o2val := reflect.New(reflect.TypeOf(o))
+	err = pr.Scan(o2val.Interface())
+	require.NoError(t, err)
+
+	return o2val.Elem().Interface()
+}
+
+func TestWriteRead(t *testing.T) {
```

Comment:
not sure if this is the right spot for these tests, but wanted to add high-level tests specific to basic field-typing cases. If you run these tests without the fixes in this pr, you will see the issues I was running into.